### PR TITLE
Implement getProduct

### PR DESCRIPTION
**Pull request by andrzejewsky**

### CODE REVIEW

File: packages/commercetools/theme/pages/index.vue [hunk 1]
```diff
@@ -155,12 +155,23 @@ import {
...
+  return encodeURIComponent(text).replace(/[!'()~*]/g, (c) => {
+    return '%' + c.charCodeAt(0).toString(16);
+  });
+}
 
 export default {
   name: "Home",
   transition: 'fade',
   data() {
+    getProduct({ variables: {
```
Comment (andrzejewsky) [inline packages/commercetools/theme/pages/index.vue]: @filrak example how to use it, I'll implement a generic helpers (eg. `getWhereByCatId`, `getWhereByCatSlug` etc.) to generate `where` param

File: packages/commercetools/api-client/src/api/getProduct/defaultQuery.ts [hunk 2]
```diff
@@ -0,0 +1,60 @@
+import gql from 'graphql-tag'
```
Comment (filrak) [inline packages/commercetools/api-client/src/api/getProduct/defaultQuery.ts:1]: can't we just use gql files?

File: packages/commercetools/api-client/src/helpers/createCommerceToolsLink/getAccessToken.ts [hunk 3]
```diff
@@ -1,7 +1,7 @@
 import SdkAuth from '@commercetools/sdk-auth'
-import { ApiConfig } from '../../types/setup'
-import { storeToken, getToken } from './tokenCache'
-import { Token } from '../../types/setup'
+import { ApiConfig } from 'api-client/src/types/setup'
```
Comment (filrak) [inline packages/commercetools/api-client/src/helpers/createCommerceToolsLink/getAccessToken.ts:2]: maybe it's better to just use '@' alias for root as common convention

File: packages/commercetools/api-client/src/helpers/i18n/index.ts [hunk 4]
```diff
@@ -0,0 +1,5 @@
+const getCurrency = () => 'USD'
```
Comment (filrak) [inline packages/commercetools/api-client/src/helpers/i18n/index.ts]: Imho it's better to keep it from config passed into `setup` then in the app you can set reference to currency and locale fields. It's also better to keep all configuration in one place

**Review by filrak — CHANGES_REQUESTED** [review]

**Pull request merged**